app/services/os_version_updater.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# A service class to update OS version for each policy with assigned hosts
+class OsVersionUpdater
+  class << self
+    def run!
+      Policy.find_each(&:update_os_versions)
```

Comment:
Yeah, is there a more optimal way to do this?

Reply:
We can simplify this by setting it up based on the supported matrix.

Here is the idea:
1. Pick one of the supported OS versions (e.g. 7.9) and its supported SSG record
2. On every non-canonical (user) profile set OS minor matching the SSG version from the step one
3. (Optionally) Check (filter by) whether the policy has hosts with that minor version
4. Repeat

For the case of dual SSG support within one minor version, first set the OS minor on latest and then do a next pass. In the next pass find profiles that are matching and set the OS, but only if the OS minor isn't already set on other profile.

Depending on the amount of data flow and performacne we might skip option 3 and leave some policies with OS minor set even if there is no such host. That might be acceptable, but we might need to discuss it. If hosts has to be checked, then they should be do so right in the profile search from step 2 (using subselects, if that's possible).

Or, we might not do any migration and should let the customers resave/update existing policies to properly assign the OS versions to profiles.

I'd suggest to split this data migration concern, as it might need a bit more discussions and testing.

Reply:
I removed the migration from this PR